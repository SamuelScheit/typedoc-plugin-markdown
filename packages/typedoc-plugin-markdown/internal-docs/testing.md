---
title: Testing guide
description: Guidance of running tests and creating fixtures.
---

# Testing

## Running tests

```bash
npm run test
```

If you require to update the snapshots run the following command:

```bash
npm run test:update
```

## Test Fixtures

A series of snapshot tests are used to ensure the output of the plugin is consistent.

- Example code for fixtures are located at `/test/fixtures/src`.
- Fixtures output scenarios are setup using a config file (`/test/fixtures/config.ts`) that provides a mechanism to easily provide a combination of options.
