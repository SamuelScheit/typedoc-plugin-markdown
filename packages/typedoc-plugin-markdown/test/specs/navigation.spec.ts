import { expectFileToEqual } from '@devtools/testing';

describe(`Navigation`, () => {
  test(`should gets Navigation Json for single entry point`, () => {
    expectFileToEqual('reflections', ['members', 'modules'], 'sidebar.json', 1);
  });

  test(`should gets Navigation Json for multiple entry points`, () => {
    expectFileToEqual('groups', ['members', 'modules'], 'sidebar.json');
  });

  test(`should get Navigation Json for packages`, () => {
    expectFileToEqual('packages', ['members', 'modules'], 'sidebar.json');
  });

  test(`should gets Navigation Json for entry modules`, () => {
    expectFileToEqual('entryfiles', ['members', 'modules'], 'sidebar.json');
  });

  test(`should gets Navigation Json for modules with parts`, () => {
    expectFileToEqual('modules', ['members', 'modules'], 'sidebar.json');
  });
});
