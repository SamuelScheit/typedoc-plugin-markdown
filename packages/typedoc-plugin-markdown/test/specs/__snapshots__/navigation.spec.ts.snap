// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should get Navigation Json for packages: (Output File Strategy "members") (Option Group "1") 1`] = `
"[
  {
    "title": "package-1",
    "children": [
      {
        "title": "Interfaces",
        "children": [
          {
            "title": "PackageInterface",
            "url": "package-1/interfaces/PackageInterface.md"
          }
        ]
      }
    ],
    "url": "package-1/index.md"
  },
  {
    "title": "package-1b",
    "children": [
      {
        "title": "Interfaces",
        "children": [
          {
            "title": "PackageInterface",
            "url": "package-1b/interfaces/PackageInterface.md"
          }
        ]
      }
    ],
    "url": "package-1b/index.md"
  },
  {
    "title": "package-2",
    "children": [
      {
        "title": "module-1",
        "url": "package-2/module-1/index.md",
        "children": [
          {
            "title": "Interfaces",
            "children": [
              {
                "title": "PackageInterface1",
                "url": "package-2/module-1/interfaces/PackageInterface1.md"
              }
            ]
          }
        ]
      },
      {
        "title": "module-2",
        "url": "package-2/module-2/index.md",
        "children": [
          {
            "title": "Interfaces",
            "children": [
              {
                "title": "PackageInterface2",
                "url": "package-2/module-2/interfaces/PackageInterface2.md"
              }
            ]
          }
        ]
      }
    ],
    "url": "package-2/index.md"
  },
  {
    "title": "package-3",
    "children": [
      {
        "title": "index",
        "url": "package-3/index/index.md",
        "children": [
          {
            "title": "Interfaces",
            "children": [
              {
                "title": "PackageInterfaceForIndexModule",
                "url": "package-3/index/interfaces/PackageInterfaceForIndexModule.md"
              }
            ]
          }
        ]
      },
      {
        "title": "module-1",
        "url": "package-3/module-1/index.md",
        "children": [
          {
            "title": "Interfaces",
            "children": [
              {
                "title": "PackageInterface1",
                "url": "package-3/module-1/interfaces/PackageInterface1.md"
              }
            ]
          }
        ]
      },
      {
        "title": "module-2",
        "url": "package-3/module-2/index.md",
        "children": [
          {
            "title": "Interfaces",
            "children": [
              {
                "title": "PackageInterface2",
                "url": "package-3/module-2/interfaces/PackageInterface2.md"
              }
            ]
          }
        ]
      }
    ],
    "url": "package-3/index.md"
  },
  {
    "title": "package-4",
    "children": [
      {
        "title": "Interfaces",
        "children": [
          {
            "title": "PackageInterface",
            "url": "package-4/interfaces/PackageInterface.md"
          }
        ]
      }
    ],
    "url": "package-4/index.md"
  }
]"
`;

exports[`Navigation should get Navigation Json for packages: (Output File Strategy "modules") (Option Group "1") 1`] = `
"[
  {
    "title": "package-1",
    "children": [
      {
        "title": "Globals",
        "url": "package-1/globals.md"
      }
    ],
    "url": "package-1/index.md"
  },
  {
    "title": "package-1b",
    "children": [],
    "url": "package-1b/index.md"
  },
  {
    "title": "package-2",
    "children": [
      {
        "title": "module-1",
        "url": "package-2/module-1.md"
      },
      {
        "title": "module-2",
        "url": "package-2/module-2.md"
      }
    ],
    "url": "package-2/index.md"
  },
  {
    "title": "package-3",
    "children": [
      {
        "title": "index",
        "url": "package-3/module.index.md",
        "children": [
          {
            "title": "Interfaces",
            "children": [
              {
                "title": "PackageInterfaceForIndexModule",
                "url": "package-3/interfaces/PackageInterfaceForIndexModule.md"
              }
            ]
          }
        ]
      },
      {
        "title": "module-1",
        "url": "package-3/module-1.md",
        "children": [
          {
            "title": "Interfaces",
            "children": [
              {
                "title": "PackageInterface1",
                "url": "package-3/interfaces/PackageInterface1.md"
              }
            ]
          }
        ]
      },
      {
        "title": "module-2",
        "url": "package-3/module-2.md",
        "children": [
          {
            "title": "Interfaces",
            "children": [
              {
                "title": "PackageInterface2",
                "url": "package-3/interfaces/PackageInterface2.md"
              }
            ]
          }
        ]
      }
    ],
    "url": "package-3/index.md"
  },
  {
    "title": "package-4",
    "children": [
      {
        "title": "Interfaces",
        "children": [
          {
            "title": "PackageInterface",
            "url": "package-4/interfaces/PackageInterface.md"
          }
        ]
      }
    ],
    "url": "package-4/index.md"
  }
]"
`;

exports[`Navigation should gets Navigation Json for entry modules: (Output File Strategy "members") (Option Group "1") 1`] = `
"[
  {
    "title": "module-1",
    "url": "module-1/README.md",
    "children": [
      {
        "title": "Classes",
        "children": [
          {
            "title": "ModuleClass",
            "url": "module-1/classes/ModuleClass.md"
          }
        ]
      }
    ]
  },
  {
    "title": "module-2",
    "url": "module-2/README.md",
    "children": [
      {
        "title": "Classes",
        "children": [
          {
            "title": "ModuleClass",
            "url": "module-2/classes/ModuleClass.md"
          }
        ]
      }
    ]
  }
]"
`;

exports[`Navigation should gets Navigation Json for entry modules: (Output File Strategy "members") (Option Group "2") 1`] = `
"[
  {
    "title": "module-1",
    "url": "module-1/index.md",
    "children": [
      {
        "title": "Classes",
        "children": [
          {
            "title": "ModuleClass",
            "url": "module-1/classes/ModuleClass.md"
          }
        ]
      }
    ]
  },
  {
    "title": "module-2",
    "url": "module-2/index.md",
    "children": [
      {
        "title": "Classes",
        "children": [
          {
            "title": "ModuleClass",
            "url": "module-2/classes/ModuleClass.md"
          }
        ]
      }
    ]
  }
]"
`;

exports[`Navigation should gets Navigation Json for entry modules: (Output File Strategy "modules") (Option Group "1") 1`] = `
"[
  {
    "title": "module-1",
    "url": "module-1.md"
  },
  {
    "title": "module-2",
    "url": "module-2.md"
  }
]"
`;

exports[`Navigation should gets Navigation Json for entry modules: (Output File Strategy "modules") (Option Group "2") 1`] = `
"[
  {
    "title": "module-1",
    "url": "module-1.md"
  },
  {
    "title": "module-2",
    "url": "module-2.md"
  }
]"
`;

exports[`Navigation should gets Navigation Json for multiple entry points: (Output File Strategy "members") (Option Group "1") 1`] = `
"[
  {
    "title": "42-module-name",
    "url": "42-module-name/index.md"
  },
  {
    "title": "<internal>",
    "url": "-internal-/index.md"
  },
  {
    "title": "basic",
    "url": "basic/index.md",
    "children": [
      {
        "title": "Enumerations",
        "children": [
          {
            "title": "EnumA",
            "url": "basic/enumerations/EnumA.md"
          },
          {
            "title": "EnumB",
            "url": "basic/enumerations/EnumB.md"
          }
        ]
      },
      {
        "title": "Classes",
        "children": [
          {
            "title": "ClassA",
            "url": "basic/classes/ClassA.md"
          },
          {
            "title": "ClassB",
            "url": "basic/classes/ClassB.md"
          }
        ]
      },
      {
        "title": "Interfaces",
        "children": [
          {
            "title": "InterfaceA",
            "url": "basic/interfaces/InterfaceA.md"
          },
          {
            "title": "InterfaceB",
            "url": "basic/interfaces/InterfaceB.md"
          }
        ]
      },
      {
        "title": "Type Aliases",
        "children": [
          {
            "title": "TypeA",
            "url": "basic/type-aliases/TypeA.md"
          },
          {
            "title": "TypeB",
            "url": "basic/type-aliases/TypeB.md"
          }
        ]
      }
    ]
  },
  {
    "title": "has-categories",
    "url": "has-categories/index.md",
    "children": [
      {
        "title": "Enumerations",
        "children": [
          {
            "title": "Category A",
            "children": [
              {
                "title": "CategoryAEnum1",
                "url": "has-categories/enumerations/CategoryAEnum1.md"
              },
              {
                "title": "CategoryAEnum2",
                "url": "has-categories/enumerations/CategoryAEnum2.md"
              }
            ]
          },
          {
            "title": "Category B",
            "children": [
              {
                "title": "CategoryBEnum1",
                "url": "has-categories/enumerations/CategoryBEnum1.md"
              },
              {
                "title": "CategoryBEnum2",
                "url": "has-categories/enumerations/CategoryBEnum2.md"
              }
            ]
          },
          {
            "title": "Other",
            "children": [
              {
                "title": "UnCategorizedEnum",
                "url": "has-categories/enumerations/UnCategorizedEnum.md"
              }
            ]
          }
        ]
      },
      {
        "title": "Interfaces",
        "children": [
          {
            "title": "Category A",
            "children": [
              {
                "title": "CategoryAInterface1",
                "url": "has-categories/interfaces/CategoryAInterface1.md"
              },
              {
                "title": "CategoryAInterface2",
                "url": "has-categories/interfaces/CategoryAInterface2.md"
              }
            ]
          },
          {
            "title": "Category B",
            "children": [
              {
                "title": "CategoryBInterface1",
                "url": "has-categories/interfaces/CategoryBInterface1.md"
              },
              {
                "title": "CategoryBInterface2",
                "url": "has-categories/interfaces/CategoryBInterface2.md"
              }
            ]
          },
          {
            "title": "Other",
            "children": [
              {
                "title": "UnCategorizedInterace",
                "url": "has-categories/interfaces/UnCategorizedInterace.md"
              }
            ]
          }
        ]
      }
    ]
  },
  {
    "title": "has-namespaces",
    "url": "has-namespaces/index.md",
    "children": [
      {
        "title": "Namespaces",
        "children": [
          {
            "title": "NamespaceB",
            "url": "has-namespaces/namespaces/NamespaceB/index.md"
          },
          {
            "title": "_Namespace_A_",
            "url": "has-namespaces/namespaces/Namespace_A_/index.md",
            "children": [
              {
                "title": "Namespaces",
                "children": [
                  {
                    "title": "NestedNamespace",
                    "url": "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace/index.md",
                    "children": [
                      {
                        "title": "Interfaces",
                        "children": [
                          {
                            "title": "NamespaceInterface",
                            "url": "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace/interfaces/NamespaceInterface.md"
                          }
                        ]
                      }
                    ]
                  }
                ]
              },
              {
                "title": "Interfaces",
                "children": [
                  {
                    "title": "NamespaceInterface",
                    "url": "has-namespaces/namespaces/Namespace_A_/interfaces/NamespaceInterface.md"
                  }
                ]
              }
            ]
          }
        ]
      },
      {
        "title": "Enumerations",
        "children": [
          {
            "title": "EnumB",
            "url": "has-namespaces/enumerations/EnumB.md"
          },
          {
            "title": "_Enum_A_",
            "url": "has-namespaces/enumerations/Enum_A_.md"
          }
        ]
      },
      {
        "title": "Classes",
        "children": [
          {
            "title": "ClassB",
            "url": "has-namespaces/classes/ClassB.md"
          },
          {
            "title": "__Class_A_",
            "url": "has-namespaces/classes/Class_A_.md"
          }
        ]
      },
      {
        "title": "Interfaces",
        "children": [
          {
            "title": "InterfaceB",
            "url": "has-namespaces/interfaces/InterfaceB.md"
          },
          {
            "title": "_Interface_A_",
            "url": "has-namespaces/interfaces/Interface_A_.md"
          }
        ]
      },
      {
        "title": "Type Aliases",
        "children": [
          {
            "title": "TypeB",
            "url": "has-namespaces/type-aliases/TypeB.md"
          },
          {
            "title": "_Type_A_",
            "url": "has-namespaces/type-aliases/Type_A_.md"
          }
        ]
      },
      {
        "title": "Variables",
        "children": [
          {
            "title": "_variable_A_",
            "url": "has-namespaces/variables/variable_A_.md"
          },
          {
            "title": "variableB",
            "url": "has-namespaces/variables/variableB.md"
          }
        ]
      },
      {
        "title": "Functions",
        "children": [
          {
            "title": "_function_A_",
            "url": "has-namespaces/functions/function_A_.md"
          },
          {
            "title": "functionB",
            "url": "has-namespaces/functions/functionB.md"
          }
        ]
      }
    ]
  },
  {
    "title": "has-references",
    "url": "has-references/index.md",
    "children": [
      {
        "title": "Functions",
        "children": [
          {
            "title": "defaultFunction",
            "url": "has-references/functions/defaultFunction.md"
          }
        ]
      }
    ]
  },
  {
    "title": "has-same-exports",
    "url": "has-same-exports/index.md",
    "children": [
      {
        "title": "Classes",
        "children": [
          {
            "title": "SomeClass",
            "url": "has-same-exports/classes/SomeClass.md"
          },
          {
            "title": "someClass",
            "url": "has-same-exports/classes/someClass-1.md"
          }
        ]
      }
    ]
  },
  {
    "title": "index",
    "url": "index/index.md",
    "children": [
      {
        "title": "Classes",
        "children": [
          {
            "title": "SomeClass",
            "url": "index/classes/SomeClass.md"
          }
        ]
      }
    ]
  }
]"
`;

exports[`Navigation should gets Navigation Json for multiple entry points: (Output File Strategy "members") (Option Group "2") 1`] = `
"[
  {
    "title": "42-module-name",
    "url": "42-module-name/index.md"
  },
  {
    "title": "<internal>",
    "url": "-internal-/index.md"
  },
  {
    "title": "basic",
    "url": "basic/index.md",
    "children": [
      {
        "title": "EnumA",
        "url": "basic/enumerations/EnumA.md"
      },
      {
        "title": "EnumB",
        "url": "basic/enumerations/EnumB.md"
      },
      {
        "title": "ClassA",
        "url": "basic/classes/ClassA.md"
      },
      {
        "title": "ClassB",
        "url": "basic/classes/ClassB.md"
      },
      {
        "title": "InterfaceA",
        "url": "basic/interfaces/InterfaceA.md"
      },
      {
        "title": "InterfaceB",
        "url": "basic/interfaces/InterfaceB.md"
      },
      {
        "title": "TypeA",
        "url": "basic/type-aliases/TypeA.md"
      },
      {
        "title": "TypeB",
        "url": "basic/type-aliases/TypeB.md"
      }
    ]
  },
  {
    "title": "has-categories",
    "url": "has-categories/index.md",
    "children": [
      {
        "title": "Category A",
        "children": [
          {
            "title": "CategoryAEnum1",
            "url": "has-categories/enumerations/CategoryAEnum1.md"
          },
          {
            "title": "CategoryAEnum2",
            "url": "has-categories/enumerations/CategoryAEnum2.md"
          },
          {
            "title": "CategoryAInterface1",
            "url": "has-categories/interfaces/CategoryAInterface1.md"
          },
          {
            "title": "CategoryAInterface2",
            "url": "has-categories/interfaces/CategoryAInterface2.md"
          }
        ]
      },
      {
        "title": "Category B",
        "children": [
          {
            "title": "CategoryBEnum1",
            "url": "has-categories/enumerations/CategoryBEnum1.md"
          },
          {
            "title": "CategoryBEnum2",
            "url": "has-categories/enumerations/CategoryBEnum2.md"
          },
          {
            "title": "CategoryBInterface1",
            "url": "has-categories/interfaces/CategoryBInterface1.md"
          },
          {
            "title": "CategoryBInterface2",
            "url": "has-categories/interfaces/CategoryBInterface2.md"
          }
        ]
      },
      {
        "title": "Other",
        "children": [
          {
            "title": "UnCategorizedEnum",
            "url": "has-categories/enumerations/UnCategorizedEnum.md"
          },
          {
            "title": "UnCategorizedInterace",
            "url": "has-categories/interfaces/UnCategorizedInterace.md"
          }
        ]
      }
    ]
  },
  {
    "title": "has-namespaces",
    "url": "has-namespaces/index.md",
    "children": [
      {
        "title": "NamespaceB",
        "url": "has-namespaces/namespaces/NamespaceB/index.md"
      },
      {
        "title": "_Namespace_A_",
        "url": "has-namespaces/namespaces/Namespace_A_/index.md",
        "children": [
          {
            "title": "NestedNamespace",
            "url": "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace/index.md",
            "children": [
              {
                "title": "NamespaceInterface",
                "url": "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace/interfaces/NamespaceInterface.md"
              }
            ]
          },
          {
            "title": "NamespaceInterface",
            "url": "has-namespaces/namespaces/Namespace_A_/interfaces/NamespaceInterface.md"
          }
        ]
      },
      {
        "title": "EnumB",
        "url": "has-namespaces/enumerations/EnumB.md"
      },
      {
        "title": "_Enum_A_",
        "url": "has-namespaces/enumerations/Enum_A_.md"
      },
      {
        "title": "ClassB",
        "url": "has-namespaces/classes/ClassB.md"
      },
      {
        "title": "__Class_A_",
        "url": "has-namespaces/classes/Class_A_.md"
      },
      {
        "title": "InterfaceB",
        "url": "has-namespaces/interfaces/InterfaceB.md"
      },
      {
        "title": "_Interface_A_",
        "url": "has-namespaces/interfaces/Interface_A_.md"
      },
      {
        "title": "TypeB",
        "url": "has-namespaces/type-aliases/TypeB.md"
      },
      {
        "title": "_Type_A_",
        "url": "has-namespaces/type-aliases/Type_A_.md"
      },
      {
        "title": "_variable_A_",
        "url": "has-namespaces/variables/variable_A_.md"
      },
      {
        "title": "variableB",
        "url": "has-namespaces/variables/variableB.md"
      },
      {
        "title": "_function_A_",
        "url": "has-namespaces/functions/function_A_.md"
      },
      {
        "title": "functionB",
        "url": "has-namespaces/functions/functionB.md"
      }
    ]
  },
  {
    "title": "has-references",
    "url": "has-references/index.md",
    "children": [
      {
        "title": "defaultFunction",
        "url": "has-references/functions/defaultFunction.md"
      }
    ]
  },
  {
    "title": "has-same-exports",
    "url": "has-same-exports/index.md",
    "children": [
      {
        "title": "SomeClass",
        "url": "has-same-exports/classes/SomeClass.md"
      },
      {
        "title": "someClass",
        "url": "has-same-exports/classes/someClass-1.md"
      }
    ]
  },
  {
    "title": "index",
    "url": "index/index.md",
    "children": [
      {
        "title": "SomeClass",
        "url": "index/classes/SomeClass.md"
      }
    ]
  }
]"
`;

exports[`Navigation should gets Navigation Json for multiple entry points: (Output File Strategy "modules") (Option Group "1") 1`] = `
"[
  {
    "title": "42-module-name",
    "url": "42-module-name.md"
  },
  {
    "title": "<internal>",
    "url": "-internal-.md"
  },
  {
    "title": "basic",
    "url": "basic.md"
  },
  {
    "title": "has-categories",
    "url": "has-categories.md"
  },
  {
    "title": "has-namespaces",
    "url": "has-namespaces/index.md",
    "children": [
      {
        "title": "Namespaces",
        "children": [
          {
            "title": "NamespaceB",
            "url": "has-namespaces/namespaces/NamespaceB.md"
          },
          {
            "title": "_Namespace_A_",
            "url": "has-namespaces/namespaces/Namespace_A_/index.md",
            "children": [
              {
                "title": "Namespaces",
                "children": [
                  {
                    "title": "NestedNamespace",
                    "url": "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace.md"
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  },
  {
    "title": "has-references",
    "url": "has-references.md"
  },
  {
    "title": "has-same-exports",
    "url": "has-same-exports.md"
  },
  {
    "title": "index",
    "url": "module.index.md"
  }
]"
`;

exports[`Navigation should gets Navigation Json for multiple entry points: (Output File Strategy "modules") (Option Group "2") 1`] = `
"[
  {
    "title": "42-module-name",
    "url": "42-module-name.md"
  },
  {
    "title": "<internal>",
    "url": "-internal-.md"
  },
  {
    "title": "basic",
    "url": "basic.md"
  },
  {
    "title": "has-categories",
    "url": "has-categories.md",
    "children": []
  },
  {
    "title": "has-namespaces",
    "url": "has-namespaces/index.md",
    "children": [
      {
        "title": "NamespaceB",
        "url": "has-namespaces/namespaces/NamespaceB.md"
      },
      {
        "title": "_Namespace_A_",
        "url": "has-namespaces/namespaces/Namespace_A_/index.md",
        "children": [
          {
            "title": "NestedNamespace",
            "url": "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace.md"
          }
        ]
      }
    ]
  },
  {
    "title": "has-references",
    "url": "has-references.md"
  },
  {
    "title": "has-same-exports",
    "url": "has-same-exports.md"
  },
  {
    "title": "index",
    "url": "module.index.md"
  }
]"
`;

exports[`Navigation should gets Navigation Json for single entry point: (Output File Strategy "members") (Option Group "1") 1`] = `
"[
  {
    "title": "Enumerations",
    "children": [
      {
        "title": "BasicEnum",
        "url": "enumerations/BasicEnum.md"
      },
      {
        "title": "EnumWithValues",
        "url": "enumerations/EnumWithValues.md"
      }
    ]
  },
  {
    "title": "Classes",
    "children": [
      {
        "title": "AbstractClass",
        "url": "classes/AbstractClass.md"
      },
      {
        "title": "CallbacksOptions",
        "url": "classes/CallbacksOptions.md"
      },
      {
        "title": "ClassWithAccessors",
        "url": "classes/ClassWithAccessors.md"
      },
      {
        "title": "ClassWithConstructorOverloads",
        "url": "classes/ClassWithConstructorOverloads.md"
      },
      {
        "title": "ClassWithModifiers",
        "url": "classes/ClassWithModifiers.md"
      },
      {
        "title": "ClassWithTypeParameters",
        "url": "classes/ClassWithTypeParameters.md"
      },
      {
        "title": "DerivedClassA",
        "url": "classes/DerivedClassA.md"
      },
      {
        "title": "DerivedClassB",
        "url": "classes/DerivedClassB.md"
      },
      {
        "title": "DisposableClass",
        "url": "classes/DisposableClass.md"
      }
    ]
  },
  {
    "title": "Interfaces",
    "children": [
      {
        "title": "BasicInterface",
        "url": "interfaces/BasicInterface.md"
      },
      {
        "title": "CustomEventInterface",
        "url": "interfaces/CustomEventInterface.md"
      },
      {
        "title": "ExtendedInterface",
        "url": "interfaces/ExtendedInterface.md"
      },
      {
        "title": "IndexableInterface",
        "url": "interfaces/IndexableInterface.md"
      },
      {
        "title": "InterfaceWithEventProperties",
        "url": "interfaces/InterfaceWithEventProperties.md"
      },
      {
        "title": "InterfaceWithTypeParameters",
        "url": "interfaces/InterfaceWithTypeParameters.md"
      }
    ]
  },
  {
    "title": "Type Aliases",
    "children": [
      {
        "title": "ArrayOfStuff",
        "url": "type-aliases/ArrayOfStuff.md"
      },
      {
        "title": "ArrayType",
        "url": "type-aliases/ArrayType.md"
      },
      {
        "title": "ConditionalType",
        "url": "type-aliases/ConditionalType.md"
      },
      {
        "title": "ExternalReferenceType",
        "url": "type-aliases/ExternalReferenceType.md"
      },
      {
        "title": "FunctionType",
        "url": "type-aliases/FunctionType.md"
      },
      {
        "title": "IndexAccessType",
        "url": "type-aliases/IndexAccessType.md"
      },
      {
        "title": "IntersectionType",
        "url": "type-aliases/IntersectionType.md"
      },
      {
        "title": "LiteralType",
        "url": "type-aliases/LiteralType.md"
      },
      {
        "title": "PartialMappedType",
        "url": "type-aliases/PartialMappedType.md"
      },
      {
        "title": "PrimitiveType",
        "url": "type-aliases/PrimitiveType.md"
      },
      {
        "title": "PromiseTypeWithObject",
        "url": "type-aliases/PromiseTypeWithObject.md"
      },
      {
        "title": "PromiseTypeWithSymbol",
        "url": "type-aliases/PromiseTypeWithSymbol.md"
      },
      {
        "title": "ReadonlyMappedType",
        "url": "type-aliases/ReadonlyMappedType.md"
      },
      {
        "title": "StringLiteralType",
        "url": "type-aliases/StringLiteralType.md"
      },
      {
        "title": "TupleType",
        "url": "type-aliases/TupleType.md"
      },
      {
        "title": "TypeWithExternalSymbolLinkMapping",
        "url": "type-aliases/TypeWithExternalSymbolLinkMapping.md"
      },
      {
        "title": "TypeWithTypeParams",
        "url": "type-aliases/TypeWithTypeParams.md"
      },
      {
        "title": "UnionType",
        "url": "type-aliases/UnionType.md"
      }
    ]
  },
  {
    "title": "Variables",
    "children": [
      {
        "title": "objectLiteralVariable",
        "url": "variables/objectLiteralVariable.md"
      },
      {
        "title": "objectWithSymbol",
        "url": "variables/objectWithSymbol.md"
      },
      {
        "title": "stringVariable",
        "url": "variables/stringVariable.md"
      },
      {
        "title": "typeOperatorVariable",
        "url": "variables/typeOperatorVariable.md"
      }
    ]
  },
  {
    "title": "Functions",
    "children": [
      {
        "title": "basicFunction",
        "url": "functions/basicFunction.md"
      },
      {
        "title": "functionReturningAFunction",
        "url": "functions/functionReturningAFunction.md"
      },
      {
        "title": "functionReturningAPromise",
        "url": "functions/functionReturningAPromise.md"
      },
      {
        "title": "functionReturningAnObject",
        "url": "functions/functionReturningAnObject.md"
      },
      {
        "title": "functionWithDefaultParameters",
        "url": "functions/functionWithDefaultParameters.md"
      },
      {
        "title": "functionWithMultipleSignatures",
        "url": "functions/functionWithMultipleSignatures.md"
      },
      {
        "title": "functionWithNamedParams",
        "url": "functions/functionWithNamedParams.md"
      },
      {
        "title": "functionWithNestedParameters",
        "url": "functions/functionWithNestedParameters.md"
      },
      {
        "title": "functionWithOptionalParameters",
        "url": "functions/functionWithOptionalParameters.md"
      },
      {
        "title": "functionWithTypeParameters",
        "url": "functions/functionWithTypeParameters.md"
      }
    ]
  }
]"
`;

exports[`Navigation should gets Navigation Json for single entry point: (Output File Strategy "modules") (Option Group "1") 1`] = `"[]"`;
